install_process.go

```diff
@@ -52,9 +53,30 @@ func (ip YarnInstallProcess) ShouldRun(workingDir string, metadata map[string]in
 	ip.logger.Action("yarn.lock -> Found")
 	ip.logger.Break()
 
-	sum, err := ip.summer.Sum(filepath.Join(workingDir, "yarn.lock"))
+	buffer := bytes.NewBuffer(nil)
+	err = ip.executable.Execute(pexec.Execution{
+		Args:   []string{"config", "list", "--silent"},
+		Stdout: buffer,
+		Stderr: buffer,
+	})
+	if err != nil {
+		return true, "", fmt.Errorf("failed to execute yarn config output:\n%s\nerror: %s", buffer.String(), err)
+	}
+
+	configFile, err := createTempFileAndWriteContents(buffer.Bytes(), "config-file")
+	if err != nil {
+		return true, "", err
+	}
+
+	nodeEnv := os.Getenv("NODE_ENV")
+	nodeEnvFile, err := createTempFileAndWriteContents([]byte(nodeEnv), "node-env")
```

Comment:
If the purpose is to simply calculate the checksum, could you append this data to the same file to avoid redundant I/O?